com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameProjectBuilder.java

```diff
@@ -443,30 +443,38 @@ private ManifestBuilder prepareManifestBuilder(ResourceNode rootNode, List<Strin
                     throw new IOException(String.format("Couldn't find public key at either: '%s' or '%s'", publicKeyFilepath, altPublicKeyFilepath));
                 }
                 publicKeyFilepath = altPublicKeyFilepath;
-
             }
         }
-
-        // If loading supplied keys failed or none were supplied, generate them instead.
-        if (privateKeyFilepath.isEmpty() || publicKeyFilepath.isEmpty()) {
-            if (project.option("liveupdate", "false").equals("true")) {
-                System.err.println("\nWarning! No public or private key for manifest signing set in liveupdate settings, generating keys instead.");
+        // Try manifest signing keys specified in project options
+        else {
+            privateKeyFilepath = project.option("manifest-private-key", "");
+            if (!privateKeyFilepath.isEmpty() && !Files.exists(Paths.get(privateKeyFilepath))) {
+                throw new IOException(String.format("Couldn't find private key at: '%s'", privateKeyFilepath));
             }
-            File privateKeyFileHandle = File.createTempFile("defold.private_", ".der");
-            privateKeyFileHandle.deleteOnExit();
+            publicKeyFilepath = project.option("manifest-public-key", "");
+            if (!publicKeyFilepath.isEmpty() && !Files.exists(Paths.get(publicKeyFilepath))) {
+                throw new IOException(String.format("Couldn't find public key at: '%s'", publicKeyFilepath));
+            }
+        }
 
-            File publicKeyFileHandle = File.createTempFile("defold.public_", ".der");
-            publicKeyFileHandle.deleteOnExit();
+        // If no keys were provided, generate them instead.
+        if (privateKeyFilepath.isEmpty() || publicKeyFilepath.isEmpty()) {
+            File privateKeyFileHandle = Paths.get(project.getRootDirectory(), "manifest.private.der").toFile();
+            File publicKeyFileHandle = Paths.get(project.getRootDirectory(), "manifest.public.der").toFile();
```

Comment:
Perhaps `ManifestBuilder.CryptographicOperations` should have these as constants?

Reply:
You think so? I'm not so sure. I think I'd prefer to keep the filenames closer to where they are actually used (ie in GameProjectBuilder). Although the full file paths are passed to `ManifestBuilder.CryptographicOperations.generateKeyPair()`. Hmm.